crates/sui-types/src/gas.rs

```diff
@@ -113,6 +114,19 @@ impl GasCostSummary {
             storage_rebate: storage_rebates.iter().sum(),
         }
     }
+
+    pub fn normalize(&self) -> GasCostSummary {
```

Comment:
I think (but am not sure) this will not play well with the conservation checks in https://github.com/MystenLabs/sui/commit/816b0144bce9ae0bf2da2402d6a65d3dce21074a. In particular, if we want to check that the sum of the `storage_rebate` field of all objects touched by the tx is equal to `summary.storage_cost * storage_rebate_discount`, that will no longer be true if we do normalization.

Reply:
well normalization has some issues as we talked in the past, all fixable (as pointed out by @lxfind) but it's also true that if we do not check this in it is not a big deal, at least in my opinion. 
It is in fact, good information to see in my opinion. 
I'll double check that condition to be sure and not get this in even if approved if something breaks there